src/Form/Form.js

```diff
@@ -18,6 +18,18 @@ function findFieldOption(options, field) {
   });
 }
 
+function mergeNewModel(stateModel, currentModelDefinition, nextModelDefinition) {
+  let newModel = {};
+
+  Object.keys(nextModelDefinition).forEach(function (key) {
+    if (currentModelDefinition[key] !== nextModelDefinition[key]) {
```

Comment:
I think this will be a more solid check:
`if (!currentModelDefinition.hasOwnProperty(key) && nextModelDefinition.hasOwnProperty(key)) {`

However, it depends on the functionality you need here. Do you want it to override if current model is `null` and next is not and when current has a value, say `'value'`, and next is `null`, and when current is empty string and next is not?